module-2/backend/week_three.md

```diff
@@ -10,14 +10,72 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 1. What is the entry at the command line to create a new rails app?
+
+  `rails new _version_ project_name -T --skip-spring --skip-turbolinks --database=postgresql`
+  
 2. What do Models generally inherit from in rails?
+
+  ApplicationRecord or ActiveRecord::Base, depending on the rails version being used
+  
 3. What do Controllers generally inherit from in a rails project?
+
+  Application Controller
+  
 4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+
+  in config/routes.rb
+  
+  `resources :horses, only: [:show]`
+  or
+  `'GET /horses/:id', to: "horses#show"`
+
 5. What rake task is useful when looking at routes, and what information does it give you?
+
+  `rake routes`
+  
+  This tells us all of the routes currently available, how/if that route is nested in its scope, the HTTP verb & URI for that route, and the prefix for the route helper
+
 6. What is an example of a route helper? When would you use them?
+
+  `/horses` -> `horses_path`
+  
+  `/horses/:id` -> `horse_path(horse)`
+  
+  These are useful in testing and when redirecting in views or in the controller. They can save some time if  a route is within the scope of another route, so we don't have to memorize all of the syntax. I particularly like how some of the routes take an argument.
+
 7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+
+  I don't remember, and I've never used `_url` as a suffix.
+
 8. What are strong params and why are the necessary?
+
+  Strong params are a setting defined in the controller by a method that limits the access of what params come through from the user submitted information into the model/db. This is necessary to prevent malicious activity.
+
 9. What role does `form_for` play in helping us create our forms?
+
+  `form_for` is a Rails helper that generates an HTML form based on the parameters defined within its block.
+
 10. How does `form_for` know where to submit the user's input?
+
+  `form_for` takes an object or objects as an argument when initialized. The input fields in the form are labeled as attributes of the object(s). I think the route this form is at and the controller that renders it may also play a factor...I think I might need to examine this more in depth.
```

Comment:
hint: the form_for knows where to submit based on the instance variable it is passed and if that variable is a `new_record?`